src/services/outliningElementsCollector.ts

```diff
@@ -84,6 +85,7 @@ namespace ts.OutliningElementsCollector {
         let depth = 0;
         const maxDepth = 20;
         function walk(n: Node): void {
+            cancellationToken.throwIfCancellationRequested();

```

Comment:
as an option instead of ThrottledCancellationToken we can do checks only for top level declarations